cmd/infra/azure/create.go

```diff
@@ -205,9 +191,37 @@ func (o *CreateInfraOptions) Run(ctx context.Context, l logr.Logger) (*CreateInf
 		}
 		result.SubnetID = *vnet.Properties.Subnets[0].ID
 		result.VNetID = *vnet.ID
-		result.VnetName = *vnet.Name
-		l.Info("Successfully created vnet", "name", result.VnetName)
+		l.Info("Successfully created vnet", "ID", result.VNetID)
+	}
+
+	// Extract network security group ID from the VNET ID if the network security group ID was not passed through the flag or if not created
+	subnetID := o.SubnetID
+	if len(subnetID) <= 0 || len(result.SubnetID) <= 0 {
+		vnet, err := azureutil.GetVnetInfoFromVnetID(ctx, result.VNetID, subscriptionID, azureCreds)
+		if err != nil {
+			return nil, err
+		}
+
+		subnetID = *vnet.Properties.Subnets[0].ID
+		l.Info("Successfully retrieved existing subnet", "ID", subnetID)
+	}
+	result.SubnetID = subnetID
+
+	// Extract network security group ID from the VNET ID if the network security group ID was not passed through the flag
+	networkSecurityGroupID := o.NetworkSecurityGroupID
+	if len(networkSecurityGroupID) <= 0 || len(result.SecurityGroupID) <= 0 {
+		vnet, err := azureutil.GetVnetInfoFromVnetID(ctx, result.VNetID, subscriptionID, azureCreds)
```

Comment:
if both subnetID and NetworkSecurityGroupID are not set, we are going to call this `GetVnetInfoFromVnetID` API twice.
I think the check whether to call this API or not should should consider both, i.e.:
```go
if len(subnetID) == 0 || len(NetworkSecurityGroupID) == 0 {
  vnet, err := azureutil.GetVnetInfoFromVnetID(ctx, result.VNetID, subscriptionID, azureCreds)

  if len(subnetID) == 0 {
    subnetID = *vnet.Properties.Subnets[0].ID
  }

  if len(NetworkSecurityGroupID) == 0  {
    networkSecurityGroupID = *vnet.Properties.Subnets[0].Properties.NetworkSecurityGroup.ID
  }
}
```